src/components/Header.jsx

```diff
@@ -0,0 +1,129 @@
+  const [anchorElNav, setAnchorElNav] = React.useState(null);
+
+  const handleClick = (route, page) => {
+    if (currentRoute !== route) {
+      navigate(route);
+      setSelectedButton(page);
+    }
+  };
+
+  const handleOpenNavMenu = (event) => {
+    setAnchorElNav(event.currentTarget);
+  };
+
+  const handleCloseNavMenu = (route, page) => {
+    setAnchorElNav(null);
+    if (route !== null || page !== null) {
+      handleClick(route, page);
+    }
+  };
+
+  return (
+    <AppBar position="static" sx={{ bgcolor: "#ffcce1" }}>
+      <Container maxWidth="xl">
+        <Toolbar disableGutters>
+          <Box sx={{ flexGrow: 1, display: { xs: "flex", md: "none" } }}>
+            <IconButton
+              size="large"
+              aria-label="account of current user"
+              aria-controls="menu-appbar"
+              aria-haspopup="true"
+              onClick={handleOpenNavMenu}
+              color="inherit"
+            >
+              <MenuIcon />
+            </IconButton>
+            <Menu
+              id="menu-appbar"
+              anchorEl={anchorElNav}
+              anchorOrigin={{
+                vertical: "bottom",
+                horizontal: "left",
+              }}
+              keepMounted
+              transformOrigin={{
+                vertical: "top",
+                horizontal: "left",
+              }}
+              open={Boolean(anchorElNav)}
+              onClose={() => handleCloseNavMenu(null, null)}
+              sx={{
+                display: { xs: "block", md: "none" },
+                "& .MuiMenu-paper": { backgroundColor: "#ffcce1" },
+              }}
+            >
+              {pages.map(({ route, page }) => (
+                <MenuItem
+                  key={page}
+                  onClick={() => handleCloseNavMenu(route, page)}
+                  sx={{
+                    ":hover": {
+                      backgroundColor: "#fcebf2",
+                    },
+                    backgroundColor:
+                      selectedButton === page ? "#fff5d7" : undefined,
+                  }}
+                >
+                  <Typography sx={{ textAlign: "center", color: "#4b5945" }}>
+                    {page}
+                  </Typography>
+                </MenuItem>
+              ))}
+            </Menu>
+          </Box>
+          <Box sx={{ flexGrow: 1, display: { xs: "none", md: "flex" } }}>
+            {pages.map(({ route, page }) => (
+              <Button
+                key={page}
+                onClick={() => handleClick(route, page)}
+                sx={{
+                  my: 2,
+                  color: "#4b5945",
+                  display: "block",
+                  ":hover": {
+                    backgroundColor: "#fcebf2",
+                  },
+                  backgroundColor:
+                    selectedButton === page ? "#fff5d7" : undefined,
+                }}
```

Comment:
Here we are using the same styles, that are used in MenuItem component, is there any way to share those styles?
I assume I can share them by creating the styles constant, and reusing it, but is there any other way?

Reply:
Yep. The same `className` would've solved repeating styles issue. 

But also another approach: you don't necessarily need to use Button for your desktop navigation. I mean, you create a component (for example using Typography), and it has all the styles, and then you use this component both in mobile & desktop menus. Styles from MenuItem and Button are applied to this new component. 

The outcome is the same. You write styles for one element only one time.